### PR TITLE
Bump go1.22 and update bootstrap version

### PR DESCRIPTION
**Pull request by AKamyshnikova**
| Q               | A
| --------------- | ---
| Bug fix?        | [x]
| New feature?    | []
| API breaks?     | []
| Deprecations?   | []
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

Bump go version to 1.22 to fix recent go related CVE. 
Update controller-tools to 0.16.5 and use updated bootstrap image.

### CODE REVIEW

File: multi-casskop/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/cscetbon/casskop/multi-casskop
 
-go 1.21
+go 1.22
```
[the remark this answers is not in the document]
Comment (AKamyshnikova) [reply multi-casskop/go.mod:3]: It will require k8s api libs bump as well

[the remark this answers is not in the document]
Comment (cscetbon) [reply multi-casskop/go.mod:3]: is that a problem @AKamyshnikova ?

[the remark this answers is not in the document]
Comment (cscetbon) [reply multi-casskop/go.mod:3]: just saw your other comment

**Comment by cscetbon** [comment]
@AKamyshnikova A test is failing

**Comment by cscetbon** [comment]
@AKamyshnikova I think it's due to a previous libraries update if I'm not mistaking (framework probably)

**Comment by AKamyshnikova** [comment]
@cscetbon I used the same si-image, but the things worked without error locally

`docker run --rm -ti -v /Users/ann/work/contrail/operators/cassandra-k8s-operator:/go/casskop ghcr.io/cscetbon/casskop-build:latest make bundle-validate
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
echo "Generate zzz-deepcopy objects"
Generate zzz-deepcopy objects
make generate-k8s
make[1]: Entering directory '/go/casskop'
/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
make[1]: Leaving directory '/go/casskop'
/go/bin/controller-gen crd paths=./api/... output:dir=./config/crd/bases schemapatch:manifests=./config/crd/bases
make update-crds
make[1]: Entering directory '/go/casskop'
echo Update CRD - Remove protocol and set config type to object CRD
Update CRD - Remove protocol and set config type to object CRD
for crd in config/crd/bases/*.yaml; do \
	crdname=$(basename $crd); \
	end=$(expr $(grep -n ^status $crd|cut -f1 -d:) - 1); \
	cat $(echo v1-crds/$crdname|sed 's/.yaml/_crd.yaml/') > /tmp/$crdname; \
	sed -e '1,/versions/d' -e "1,${end}s/^..//" $crd >> /tmp/$crdname; \
	cp /tmp/$crdname $crd; \
	yq -i e '.spec.versions[0].storage = false' $crd; \
done
for chart in casskop multi-casskop; do \
  cp -v config/crd/bases/*.yaml charts/$chart/crds/; \
done
'config/crd/bases/db.orange.com_cassandrabackups.yaml' -> 'charts/casskop/crds/db.orange.com_cassandrabackups.yaml'
'config/crd/bases/db.orange.com_cassandraclusters.yaml' -> 'charts/casskop/crds/db.orange.com_cassandraclusters.yaml'
'config/crd/bases/db.orange.com_cassandrarestores.yaml' -> 'charts/casskop/crds/db.orange.com_cassandrarestores.yaml'
'config/crd/bases/db.orange.com_cassandrabackups.yaml' -> 'charts/multi-casskop/crds/db.orange.com_cassandrabackups.yaml'
'config/crd/bases/db.orange.com_cassandraclusters.yaml' -> 'charts/multi-casskop/crds/db.orange.com_cassandraclusters.yaml'
'config/crd/bases/db.orange.com_cassandrarestores.yaml' -> 'charts/multi-casskop/crds/db.orange.com_cassandrarestores.yaml'
make[1]: Leaving directory '/go/casskop'
operator-sdk generate kustomize manifests -q;\
/go/bin/kustomize build config/crd | operator-sdk generate bundle -q --overwrite 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
WARN[0000] ClusterServiceVersion validation: [OperationFailed] provided API should have an example annotation 
INFO[0000] Creating bundle.Dockerfile                   
INFO[0000] Creating bundle/metadata/annotations.yaml    
INFO[0000] Bundle metadata generated suceessfully       
operator-sdk bundle validate ./bundle
WARN[0000] Warning: Value db.orange.com/v1alpha1, Kind=CassandraBackup: provided API should have an example annotation 
WARN[0000] Warning: Value db.orange.com/v2, Kind=CassandraBackup: provided API should have an example annotation 
WARN[0000] Warning: Value db.orange.com/v1alpha1, Kind=CassandraCluster: provided API should have an example annotation 
WARN[0000] Warning: Value db.orange.com/v2, Kind=CassandraCluster: provided API should have an example annotation 
WARN[0000] Warning: Value db.orange.com/v1alpha1, Kind=CassandraRestore: provided API should have an example annotation 
WARN[0000] Warning: Value db.orange.com/v2, Kind=CassandraRestore: provided API should have an example annotation 
INFO[0000] All validation tests have completed successfully 
`
So, need to understand what can be the difference

**Review by cscetbon — APPROVED** [review]

**Comment by AKamyshnikova** [comment]
e2e tests passed on latest commit https://github.com/AKamyshnikova/casskop/actions/runs/11682452903/job/32529683560 (backup/restore failed due to s3 credentials)

**Pull request merged**